analyses/0014-vitess/vitess-analysis.md

```diff
@@ -0,0 +1,1019 @@
+
+Here's how I'd rewrite it, defining placeholders for the parameters:
+
+> ```
+> vtctldclient AddCellInfo \
+>   --root <root_dir> \
+>   --server-address <cell_topo_addr> \
+>   cell1
+> ```
+>
+> where:
+> - <root_dir> is the root directory of the server installation
+> - <cell_topo_addr> is the IP address of the topo server
+
+(or whatever the actual descriptions of the parameters are.)
+
+Here's an outline for a one-page procedure writeup:
+
+- Title ("ABCing XYZ")
+  - Context (Describe where the procedure fits in the overall use case)
+  - Prerequisites (Hardware and software requirements, dependencies, procedures that
+    must be completed first)
+  - Procedure ("To ABC an XYZ, do the following.")
+    - Step 1 (*One* CLI command, or action. Don't combine actions.)
+    - Step 2 (etc.)
+  - Result (Optional - include only if there's something remarkable about the outcome.)
+  - Next Steps (What procedure or use case typically follows this. If it depends on
+    context, explain the options.)
+
+*Reorganize*
+
+Ensure that the various sections of the User Guide cover all usage scenarios. Reorganize
+the User Guide:
+
+First, organize by user role, if there are distinct roles. As I understand it, there
+are basically three user roles in Vitess:
+- Vitess admin
+- Database admin
+- Application developer
+
+ User roles traditionally are the basis for a "User Guide":
+- Admin Guide
+- DBA Guide
+- Programmer's Guide
+- etc.
+
+Within roles, organize the tasks around use cases. Don't be afraid to split up tasks
+that use the same tool (CLI or other). In other words, pick and choose commands and
+actions that server a use case rather than trying to document everything you can do
+with the command in one place. (There is a place to exhaustively document a tool,
+but that's in the Reference, not a User Guide.)
+
+The existing Vitess User Guides are already partially organized around user roles,
+but they can be regrouped. In any case, make the user roles explicit:
+
+*Vitess admin*:
+- Configuration
+- Running in production
+- Operational
+- Migration
+
+*DBA*:
+- VSchema and Query Serving
+- SQL statement analysis
+- Making schema changes
+
+*Application programmer*:
+- Query serving
+- Making schema changes (don't duplicate the section here. Instead provide links to
+  any tasks that are identical)
+- Query optimization
+
+(Again, these are my understanding of the Vitess user roles. Adjust the details if
+they're different.
+
+**Troubleshooting**
+
+Consolidate all troubleshooting information into a Troubleshooting Guide. Or, create
+one troubleshooting guide per user guide/user role. In either case, create an escalation
+path for any problems with a task (the escalation path might be:
+*troubleshooting procedure > Slack Channel > project Issue*). Get rid of the
+[VReplication FAQ](https://vitess.io/docs/21.0/reference/vreplication/faq/)
+in the reference section and put the information in a troubleshooting section.
```

Comment:
Would you recommend consolidating everything into https://vitess.io/docs/faq/
And potentially renaming that to `Troubleshooting`?

Reply:
See recommendations in the analysis. The good news is that there's a lot of good information in the FAQ. The bad news is that it's outside your versioning regimen. Also, I'm not a fan of FAQs in general. I recommend folding the information into the regular technical documentation.